### PR TITLE
JP-1737: Propagate error through Extract1D step

### LINKED ISSUE #5373: Spec2Pipeline LRS processing: Extract1d does not produce error column (all 0) in *x1d.fits file
_Issue [JP-1737](https://jira.stsci.edu/browse/JP-1737) was created on JIRA by [Juergen Schreiber](https://jira.stsci.edu/secure/ViewProfile.jspa?name=schreibe):_

I ran Spec2Pipeline skipping the Photom step to get the 'raw' extracted spectrum.

The error column in the resulting x1d file stays at 0 although it would be easy to calculate: just take the standard deviation of the sum of each pixel row of the aperture.

### PR DESCRIPTION
**Pull request by tapastro**
Closes #5373 

Resolves [JP-1737](https://jira.stsci.edu/browse/JP-1737)

**Description**

This PR propagates error/variance arrays through the various branches of Extract1D. Some existing functions were modified out of convenience when handling many new arrays. A first pass on documentation has been completed.

Working on regression test results now.

Checklist
- [ ] Tests
- [x] Documentation
- [x] Change log
- [x] Milestone
- [x] Label(s)

### CODE REVIEW

File: jwst/extract_1d/ifu.py [hunk 1]
```diff
@@ -708,14 +848,20 @@ def image_extract_ifu(input_model, source_type, extract_params):
     """
 
     data = input_model.data
+    try:
+        var_poisson = input_model.var_poisson
+        var_rnoise = input_model.var_rnoise
+        var_flat = input_model.var_flat
+    except AttributeError:
```
[the remark this answers is not in the document]
Comment (jdavies-st) [reply jwst/extract_1d/ifu.py:855]: These will always be defined as attributes, if they are in the schema.  I.e. they will never return `AttributeError`, unless somehow `input_model` is an object that doesn't have them in the schema.  Is that a case?  `CubeModel` has them in the schema, but `IFUCubeModel` does not.  Is the input to this step always the latte for IFU?

File: jwst/extract_1d/ifu.py [hunk 2]
```diff
@@ -131,30 +139,58 @@ def ifu_extract1d(input_model, ref_dict, source_type, subtract_background,
     if input_units_are_megajanskys:
         # Convert flux from MJy to Jy, and convert background to MJy / sr.
         flux = temp_flux * 1.e6
+        f_var_poisson *= 1.e12  # MJy**2 --> Jy**2
+        f_var_rnoise *= 1.e12  # MJy**2 --> Jy**2
+        f_var_flat *= 1.e12  # MJy**2 --> Jy**2
         surf_bright[:] = 0.
-        background[:] /= pixel_solid_angle
+        sb_var_poisson = 0.
```
[the remark this answers is not in the document]
Comment (tapastro) [reply jwst/extract_1d/ifu.py:146]: That was my understanding of what was going on here with setting surf_bright to 0, so I assumed a similar result was desired for the var arrays as well.

**Comment by tapastro** [comment]
Lingering regression test issue has been resolved - the combine1d difference was caused by a missing meta.observation.bkgdtarg = 'EXTENDED' - user error in updating the new datamodel with only='PRIMARY'...

File: jwst/extract_1d/extract.py [hunk 3]
```diff
@@ -2144,11 +2200,17 @@ def log_extraction_parameters(self):
         log.debug(f"smoothing_length = {self.smoothing_length}")
         log.debug(f"position_correction = {self.position_correction}")
 
-    def extract(
-            self, data: np.ndarray, wl_array: np.ndarray, verbose: bool
-    ) -> Tuple[
-        float, float, np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray
-    ]:
+    def extract(self, data,
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply jwst/extract_1d/extract.py:2203]: Looks like this still needs fixing (add np.ndarray declaration for `data`)

File: jwst/extract_1d/extract1d.py [hunk 4]
```diff
@@ -81,9 +93,27 @@ def extract1d(image, lambdas, disp_range,
     countrate : ndarray, 1-D, float64
         The extracted spectrum in units of counts / s.
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply jwst/extract_1d/extract1d.py:94]: Still need to update these descriptions for the param list. I know it doesn't affect operation of the code, but let's not leave these lingering here for the future.

File: jwst/extract_1d/ifu.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hbushouse) [reply jwst/extract_1d/ifu.py:860]: As long as you've confirmed that this works properly when running on IFU data (i.e. it actually does throw an `AttributeError`), then I'm happy with it and can consider this resolved.

**Review by hbushouse — COMMENTED** [review]
Looks like just a couple minor requests for updates from the initial review still need to be implemented. Then it'll be good to go.

**Comment by tapastro** [comment]
> Looks like just a couple minor requests for updates from the initial review still need to be implemented. Then it'll be good to go.

Cleaned these up - I did confirm that AttributeError is triggered using the test_miri_mrs_extract1d regtest.

**Review by hbushouse — APPROVED** [review]

**Pull request merged**